src/System.Management.Automation/engine/SessionStateContent.cs

```diff
@@ -848,6 +849,10 @@ internal void ClearContent(string[] paths, bool force, bool literalPath)
 
             try
             {
+                if (Directory.Exists(path))
+                {
+                    throw PSTraceSource.NewNotSupportedException(SessionStateStrings.ClearDirectoryContent, path);
+                }
```

Comment:
I'd expect non-terminating error.

/cc @mklement0

Reply:
Yes, this should be non-terminating error.

Reply:
Also, the check should be in the FileSystemProvider in this function: https://github.com/PowerShell/PowerShell/blob/b27380dc510fe3cc2b8dbef056d882e6086dcd20/src/System.Management.Automation/namespaces/FileSystemProvider.cs#L6599 not in the engine.